### PR TITLE
Merge RecommendationServlet.java 

### PR DESCRIPTION
**Pull request by ethyus**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.mapdb.*;
+import com.google.gson.Gson;
+import java.util.Map;
+import java.util.HashMap;
+
+@WebServlet("/recommendation")
+public class RecommendationServlet extends HttpServlet {
+  private DB db;
+  private BTreeMap<String, String[]> queryNearestNeighbors;
+  private Map<String, String> jsonErrorInfo;
```
Comment (tylerhou) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java]: Why are we storing this in the class?

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  @Override
+  public void init() {
+    String path = Paths.get("/home/ngothomas/downloads/webapp/step197-2020/word2vec.db").toString();
```
Comment (tylerhou) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java]: Probably the wrong path.

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      return;
+    }
+
+    String queryWord = request.getParameter("queryWord");
+    int numOfWordsRequested = Integer.parseInt(request.getParameter("numOfWordsRequested"));
+
+    try {
+      String[] neighbors = queryNearestNeighbors.get(queryWord);
+
+      // We skip neighbors[0] because that is the same word as "queryWord"
```
Comment (tylerhou) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java:59]: Good comment.

**Review by tylerhou — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java (hunk 1 above)
Comment (ethyus) [reply src/main/java/com/google/sps/servlets/RecommendationServlet.java]: In order for the variables to be accessed by the init() and destroy() function. I only want to open the db and grab the queryNearestNeighbors one time. Though looking back, jsonErrorInfo can be put in inside the method call

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java (hunk 2 above)
Comment (ethyus) [reply src/main/java/com/google/sps/servlets/RecommendationServlet.java]: Fixed

**Pull request merged**